### PR TITLE
Support advance properteis for sql hint

### PR DESCRIPTION
**Pull request by gzzhanghao**
Add support for advance properties (`displayText` for example) for sql-hint addon.

Here is an example for the new feature:
```
options.tables = [{
  text: 'table-name',
  displayText: 'table-name | comment',
  columns: [{
    text: 'column-name',
    displayText: 'column-name | column-type'
  }]
}]
```

### CODE REVIEW

**Comment by marijnh** [comment]
You are changing the table format in an incompatible, breaking way. That is not acceptable.

**Comment by gzzhanghao** [comment]
@marijnh The patch is backward compatible, the original table format is supported as well.

**Comment by marijnh** [comment]
Fair enough. Merged as 8f13aa615b8f7 and simplified and tested in 1eb72e8ccc17412.

File: addon/hint/sql-hint.js [hunk 1]
```diff
@@ -26,9 +26,37 @@
     return CodeMirror.resolveMode(mode).keywords;
   }
 
+  function getText(item) {
+    if (typeof item == 'string') {
+      return item;
+    }
+    return item.text;
```
Comment (prathik457) [inline addon/hint/sql-hint.js]: Shouldn't this be item.displayText?
Let's suppose I have a table object {columns:{....}, text: '{{table1}}', displayText:'table1'}
Then when I type 'ta' it does not return me anything. Am I missing something?

Comment (marijnh) [reply addon/hint/sql-hint.js]: > Shouldn't this be item.displayText?

No, this is used to get the actual completion text, not the label.

**Pull request closed without merging**